announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -694,11 +703,42 @@ else if (view.equals(VIEW_MODE_PUBLIC))
 							{
 								messages = getMessagesPublic(site, channel, null, true, state, portlet);
 							}
+							else if (VIEW_MODE_BYROLE.equals(view))
+							{
+								messages = getMessagesByRoles(site, channel, null, true, state, portlet);
+							}
 						}
 						else
 						{
 							messages = getMessages(channel, null, true, state, portlet);
 						}
+						List<AnnouncementWrapper> finalMessages = new ArrayList<>();
+						for (AnnouncementWrapper message : messages) {
+							site = SiteService.getSite(EntityManager.newReference(message.getReference()).getContext());
+							String selectedRoles = (String) message.getProperties().get(SELECTED_ROLES_PROPERTY);
+							boolean addAnnounMessage = false;
+							if (selectedRoles == null) {
+								addAnnounMessage = true;
+							} else {
+								String[] selectedRolesArray = selectedRoles.replaceAll("[\\[\\]]", "").replaceAll(", ", ",").split(",");
+								Member actualMember = site.getMember(userDirectoryService.getCurrentUser().getId());
+								String memberRole = actualMember.getRole().getId();
+								for (String selectedRole : selectedRolesArray) {
+									if (selectedRole.equals(memberRole)) {
+										addAnnounMessage = true;
+									}
+								}
+								if (actualMember.getUserEid().equals(message.getAnnouncementHeader().getFrom().getDisplayId())) {
```

Comment:
It should be getId ?

the variable should be currentMember

Reply:
Ok